src/Reflection/ReflectionParameter.php

```diff
@@ -187,7 +187,10 @@ private function parseDefaultValueNode(): void
         ) {
             $this->isDefaultValueConstant   = true;
             $this->defaultValueConstantName = $defaultValueNode->name->parts[0];
-            $this->defaultValue             = null;
+            $this->defaultValue             = (new CompileNodeToValue())->__invoke(
+                $defaultValueNode,
+                new CompilerContext($this->reflector, null),
```

Comment:
I think `$this->getDeclaringClass()` should still be passed in here.

Still, assuming we get rid of `$this->defaultValue`, and instead store the AST node for the constant, we can use `CompileNodeToValue` in a getter, and defer the exception (for now)

Reply:
@Ocramius I moved ```CompileNodeToValue``` in the getter and added a ```@throw``` comment there

Reply:
Nice!